Description: Patch adapted from 3.4.x to fix new warnings that break
 the build under gcc >= 4.6.1.
Origin: https://issues.apache.org/jira/browse/ZOOKEEPER-1117

Index: zookeeper/src/c/src/zookeeper.c
===================================================================
--- a/src/c/src/zookeeper.c	(revision 1155281)
+++ b/src/c/src/zookeeper.c	(working copy)
@@ -462,7 +462,7 @@
         *port_spec = '\0';
         port_spec++;
         port = strtol(port_spec, &end_port_spec, 0);
-        if (!*port_spec || *end_port_spec) {
+        if (!*port_spec || *end_port_spec || port == 0) {
             LOG_ERROR(("invalid port in %s", host));
             errno=EINVAL;
             rc=ZBADARGUMENTS;
Index: zookeeper/src/c/src/cli.c
===================================================================
--- a/src/c/src/cli.c	(revision 1155281)
+++ b/src/c/src/cli.c	(working copy)
@@ -581,12 +581,14 @@
         FD_SET(0, &rfds);
         rc = select(fd+1, &rfds, &wfds, &efds, &tv);
         events = 0;
-        if (FD_ISSET(fd, &rfds)) {
-            events |= ZOOKEEPER_READ;
+        if (rc > 0) {
+            if (FD_ISSET(fd, &rfds)) {
+           	    events |= ZOOKEEPER_READ;
+            }
+            if (FD_ISSET(fd, &wfds)) {
+                events |= ZOOKEEPER_WRITE;
+            }
         }
-        if (FD_ISSET(fd, &wfds)) {
-            events |= ZOOKEEPER_WRITE;
-        }
         if(batchMode && processed==0){
           //batch mode
           processline(cmd);
